File: silver/BOJ11047.py
# https://www.acmicpc.net/problem/11047

import sys
input = sys.stdin.readline

n, k = map(int, input().split())
coins = [int(input()) for i in range(n)]
coins.reverse()
temp = []
answer = 0

for i in range(n) :
    if coins[i] > k :
        temp.append(coins[i])

searchIndex = len(temp)

for i in range(searchIndex, n) :
    answer += k // coins[i]
    k = k % coins[i]

print(answer)

File: BOJ/Silver/BOJ11047.java
/* https://www.acmicpc.net/problem/11047 */

import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws Exception{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int money = Integer.parseInt(st.nextToken());
        List<Integer> coins = new ArrayList<>();
        int answer = 0;

        for(int i=0; i<n; i++){
            coins.add(Integer.parseInt(br.readLine()));
        }

        Collections.sort(coins, Collections.reverseOrder());

        for(int coin : coins){
            if(coin > money){
                continue;
            }
            answer += money/coin;
            money = money % coin;
        }
        System.out.print(answer);
    }
}